fluidsynth/src/sfloader/fluid_defsfont.h

```diff
@@ -435,7 +435,7 @@ struct _fluid_defpreset_t
 {
   fluid_defpreset_t* next;
   fluid_defsfont_t* sfont;                  /* the soundfont this preset belongs to */
-  char name[21];                        /* the name of the preset */
+  char name[22];                        /* the name of the preset */
```

Comment:
> 4 bytes, but 5 decimal digits.

Really 2 bytes (unsigned short). But yes, 5 digits. By "become" I meant the actual output of the sprintf, where digits become bytes.

> Anyway, internally I would prefer to have uniquely named preset, i.e. not doing sfpreset->prenum % 128.

OK, right. If input values are valid throughout the full `unsigned short` range, then we must of course represent their values properly. But then, sadly, GCC is right: We had a buffer overflow here. We'd need to shorten the `Bank%d,Preset%d` format string by at least one byte, in order to fit into 20(+1) bytes, as per the SF2 spec. And I think you're right, we shouldn't use longer names than the SF2 spec permits (at least not only for the purpose of fitting this default string).